shanoir-ng-datasets/src/main/java/org/shanoir/ng/examination/service/ExaminationServiceImpl.java

```diff
@@ -116,15 +121,54 @@ public void deleteById(final Long id) throws ShanoirException, SolrServerExcepti
 		} else {
 			List<DatasetAcquisition> dsAcqs = examination.getDatasetAcquisitions();
 			if (dsAcqs != null) {
+				Map<String, String> eventProperties = new HashMap<>();
+				float progressMax = 0f;
 				for (DatasetAcquisition dsAcq : dsAcqs) {
-					this.datasetAcquisitionService.deleteById(dsAcq.getId());
+					if (event != null) {
+						progressMax += dsAcq.getDatasets().size();
+						eventProperties.put("progressMax", String.valueOf(progressMax));
+						event.setEventProperties(eventProperties);
+					}
+				}
+				for (DatasetAcquisition dsAcq : dsAcqs) {
+					event.setMessage("Delete examination - acquisition with id : " + dsAcq.getId());
+					eventService.publishEvent(event);
+					this.datasetAcquisitionService.deleteById(dsAcq.getId(), event);
 				}
 			}
 			examinationRepository.deleteById(id);
 		}
 		rabbitTemplate.convertAndSend(RabbitMQConfiguration.RELOAD_BIDS, objectMapper.writeValueAsString(examination.getStudyId()));
 	}
 
+	@Async
+	@Transactional
+	public void deleteExaminationAsync(Long examinationId, Long studyId, ShanoirEvent event) {
+		try {
+			String dataPath = getExtraDataFilePath(examinationId, "");
+			File fileToDelete = new File(dataPath);
+			if (fileToDelete.exists()) {
+				FileUtils.deleteDirectory(fileToDelete);
+			}
+			deleteById(examinationId, event);
+			rabbitTemplate.convertAndSend(RabbitMQConfiguration.EXAMINATION_STUDY_DELETE_QUEUE, objectMapper.writeValueAsString(event));
+			rabbitTemplate.convertAndSend(RabbitMQConfiguration.RELOAD_BIDS, objectMapper.writeValueAsString(studyId));
+
+		} catch (Exception e) {
+			event.setStatus(ShanoirEvent.ERROR);
+			event.setMessage("Error during delete of examination with id : " + examinationId);
+			event.setProgress(-1f);
+			eventService.publishEvent(event);
+			LOG.error("Error during delete of examination with id : " + examinationId);
+		} finally {
+			event.setObjectId(String.valueOf(event.getId()));
```

Comment:
This should not be in the finally.
finally block is executed even after an exception was catched. In case of exception, you should not have this code executed, you want the error to be displayed.